diff --git a/pom.xml b/pom.xml
index 0156d4c..6996c85 100644
--- a/pom.xml
+++ b/pom.xml
@@ -202,9 +202,14 @@
       <version>1.10</version>
     </dependency>
     <dependency>
-      <groupId>bouncycastle</groupId>
-      <artifactId>bcprov-jdk15</artifactId>
-      <version>140</version>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcprov-jdk18on</artifactId>
+      <version>1.78.1</version>
+    </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcpkix-jdk18on</artifactId>
+      <version>1.78.1</version>
     </dependency>
     <dependency>
       <groupId>org.apache.ant</groupId>
@@ -269,6 +274,10 @@
   </dependencies>

   <repositories>
+    <repository>
+      <id>central</id>
+      <url>https://repo1.maven.org/maven2/</url>
+    </repository>
     <repository>
       <id>repo.jenkins-ci.org</id>
       <url>https://repo.jenkins-ci.org/public/</url>
diff --git a/src/main/java/io/jenkins/update_center/Signer.java b/src/main/java/io/jenkins/update_center/Signer.java
index 84b953a..b279748 100644
--- a/src/main/java/io/jenkins/update_center/Signer.java
+++ b/src/main/java/io/jenkins/update_center/Signer.java
@@ -1,7 +1,6 @@
 package io.jenkins.update_center;

 import io.jenkins.update_center.json.JsonSignature;
-
 import io.jenkins.update_center.util.Environment;
 import org.apache.commons.codec.binary.Base64;
 import org.apache.commons.codec.binary.Hex;
@@ -9,8 +8,10 @@ import org.apache.commons.codec.digest.DigestUtils;
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.io.output.NullOutputStream;
 import org.apache.commons.io.output.TeeOutputStream;
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
-import org.bouncycastle.openssl.PEMReader;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
 import org.jvnet.hudson.crypto.CertificateUtil;
 import org.jvnet.hudson.crypto.SignatureOutputStream;
 import org.kohsuke.args4j.Option;
@@ -23,7 +24,6 @@ import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.security.DigestOutputStream;
 import java.security.GeneralSecurityException;
-import java.security.KeyPair;
 import java.security.MessageDigest;
 import java.security.PrivateKey;
 import java.security.Signature;
@@ -86,8 +86,9 @@ public class Signer {
         X509Certificate signer = certs.get(0); // the first one is the signer, and the rest is the chain to a root CA.

         PrivateKey key;
-        try (PEMReader pem = new PEMReader(Files.newBufferedReader(privateKey.toPath(), StandardCharsets.UTF_8))) {
-             key = ((KeyPair) pem.readObject()).getPrivate();
+        try (PEMParser pem = new PEMParser(Files.newBufferedReader(privateKey.toPath(), StandardCharsets.UTF_8))) {
+            PrivateKeyInfo privateKeyInfo = (PrivateKeyInfo) pem.readObject();
+            key = new JcaPEMKeyConverter().getPrivateKey(privateKeyInfo);
         }

         // the correct signature (since Jenkins 1.433); no longer generate wrong signatures for older releases.
